File: book/snippets/js/src/hello.ts
// ANCHOR: full
import {Keys, Client, EventBuilder, NostrSigner, loadWasmAsync} from "@rust-nostr/nostr-sdk";

async function hello() {
    // Load WASM
    await loadWasmAsync();

    // ANCHOR: client
    let keys: Keys = Keys.generate();
    let signer = NostrSigner.keys(keys);
    let client = new Client(signer);
    // ANCHOR_END: client

    // ANCHOR: connect
    await client.addRelay("wss://relay.damus.io")
    await client.connect();
    // ANCHOR_END: connect

    // ANCHOR: publish
    let builder = EventBuilder.textNote("Hello, rust-nostr!");
    let output = await client.sendEventBuilder(builder);
    // ANCHOR_END: publish

    // ANCHOR: output
    console.log("Event ID:", output.id.toBech32());
    console.log("Sent to:", output.success);
    console.log("Not sent to:", output.failed);
    // ANCHOR_END: output
}

hello();
// ANCHOR_END: full

File: book/snippets/kotlin/shared/src/main/kotlin/rust/nostr/snippets/Hello.kt
package rust.nostr.snippets

// ANCHOR: full
import rust.nostr.sdk.*

suspend fun hello() {
    // ANCHOR: client
    val keys = Keys.generate()
    val signer = NostrSigner.keys(keys)
    val client = Client(signer = signer)
    // ANCHOR_END: client

    // ANCHOR: connect
    client.addRelay("wss://relay.damus.io")
    client.connect()
    // ANCHOR_END: connect

    // ANCHOR: publish
    val builder = EventBuilder.textNote("Hello, rust-nostr!")
    val res = client.sendEventBuilder(builder)
    // ANCHOR_END: publish

    // ANCHOR: output
    println("Event ID: ${res.id.toBech32()}")
    println("Sent to: ${res.output.success}")
    println("Not sent to: ${res.output.failed}")
    // ANCHOR_END: output
}
// ANCHOR_END: full
